game.py

```diff
@@ -1,41 +1,82 @@
-from random import randint
+from operator import itemgetter
+
+from random import choice, randint
 
 from exceptions import NimException
 
 
 class Bot():
+    subclasses = []
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        Bot.subclasses.append(cls)
+
+    def valid_pile_indexes(self, state):
+        return [i for i, nb_stones in enumerate(state) if nb_stones >= 1]
+
     def move(self, state):
-        raise NotImplementedError
+        if not self.valid_pile_indexes(state):
+            raise NimException(f'No valid moves on state: {state}')
+        return self.move_strategy(state)
```

Comment:
We probably should add this method to the `Bot` class so it doesn't throw an `AttributeError` if someone forget to implement it when creating their own bot.